### ISSUE TITLE
issue in fasion_mnist.ipynb

### ISSUE THREAD

**Issue by gauravkoradiya** [udacity/deep-learning-v2-pytorch#86]
i used solve a problem by using sequential method in nn module but it return an error of size missmatch .I paste my code to here with error

model = nn.Sequential(nn.Linear(784, 256),
                      nn.ReLU(),
                      nn.Linear(256, 128),
                      nn.ReLU(),
                      nn.Linear(128, 64),
                      nn.ReLU(),
                      nn.Linear(64, 10),
                      nn.LogSoftmax(dim=1))

criterion = nn.NLLLoss()
optimizer = optim.Adam(model.parameters(), lr=0.001)

# TODO: Train the network here
epochs = 5

for e in range(epochs):
    running_loss = 0
    for images, labels in trainloader:
        # Flatten MNIST images into a 784 long vector
        imagees = images.view(1,-1)
        
        log_ps = model(images)
        loss = criterion(log_ps, labels)
        
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
        
        running_loss += loss.item()
    else:
        print("Training loss:",running_loss/len(trainloader))

when i run this code at that time it prodce error like : 
size mismatch, m1: [1792 x 28], m2: [784 x 256] at c:\programdata\miniconda3\conda-bld\pytorch_1532501501514\work\aten\src\th\generic/THTensorMath.cpp:2070

**Comment by miniamisha** [comment]
You wrote:
imagees = images.view(1,-1), I think there is mistake in this.
Try this:
images=images.view(images.shape[0],-1) as it means starting from index 0 image and -1 means multiplying all other dimension to make image flat.

**Comment by mukulkrjha** [comment]
@gauravkoradiya 
Short Answer - You are __Flattening__ the images in _wrong manner_.
Lets move on to error and its solution gradually by developing an intuition about it. Firstly you need a function to reshape the images tensor , and you're using `view()` function which does the job correctly. But the *parameters* ( or *dimensions* ) you passed on there needs to be changed. 
_How_ and _Why_?? 
The images tensor is having the size `[64,1,28,28]` which simply means it contains **64 images** having a single color channel(i.e **grayscale** ) and each image is having dimensions of `28 by 28` pixels (`28x28 = 784` in total) . Now we want to modify its dimensions (i.e Flatten it) But this modification in dimensions must be made in such a manner that resulting tensor will be **compatible for matrix multiplication with first layer of our Network model**. And Matrix multiplication Compatibility means that _no of columns of first matrix must match the no. of rows of second matrix_also known as **matching of inner dimensions** ( Now you maybe having some intuitions about your error, see your input is of size `[1792,28]` and first layer is of size `[784,256]` clearly 28  != 784 so is the error) .

So we will use `images = images.view( images.shape[0],784)` where the _0th_ index of images shape(images.shape[0]) gives us 64, representing no. of total images as explained above and _784_ is just used to show conversion of _2D matrix of 28x28 size to 1D tensor of size 784_.
At this point your input to NN is having size `[64,784]` and first layer of NN is having size `[784,256]`.
clearly inner dimensions of these matrix is same and no mismatch is present.
#### Note : 
> `images.view(images.shape[0],-1 )` is just a short hand representation, where **-1** basically **let _python_ itself figure out the size** of columns.
